lib/utils/router.js

```diff
@@ -16,17 +16,20 @@ define(['Navigo'], function (Navigo) {
     }
 
     function gotoNode(d) {
-      if (d.nodeId in objects.nodes) {
+      if (objects.nodeDict[d.nodeId]) {
         targets.forEach(function (t) {
-          t.gotoNode(objects.nodes[d.nodeId], objects.nodeDict);
+          t.gotoNode(objects.nodeDict[d.nodeId], objects.nodeDict);
         });
       }
     }
 
     function gotoLink(d) {
-      if (d.linkId in objects.links) {
+      var link = objects.links.find(function (value) {
```

Comment:
Idea: performance question: what happen more often?
 * foreach in setData every minute
 * gotoLink find the link

Reply:
goToLink is used then url has a link. Max 1 per minute update or by clicking a link. Lookup takes 0.1 millisecond with 1000 links.